system/handlers/formcontrols/ReadOnly.cfc

```diff
@@ -0,0 +1,19 @@
+component {
+
+	property name="presideObjectService" inject="presideObjectService";
+
+	public string function index( event, rc, prc, args={} ) {
+		var renderer = args.renderer ?: "";
+		var type     = presideObjectService.getObjectPropertyAttribute( args.sourceObject, args.name, "type" );
```

Comment:
See `/system/services/rendering/ContentRendererService.cfc$getRendererForField()`. This would be a better start and re-use existing logic in terms of finding the best default renderer for the field.